client/state/products-list/selectors.js

```diff
@@ -33,3 +44,40 @@ export function getProductDisplayCost( state, productSlug ) {
 
 	return product.cost_display;
 }
+
+export const getDiscountedOrRegularPrice = ( state, siteId, plan, isMonthlyPreference ) => {
+	const isMonthly = isMonthlyPreference && plan.term !== TERM_MONTHLY;
+	return (
+		getPlanDiscountedRawPrice( state, siteId, plan.getStoreSlug(), { isMonthly } ) ||
+		getPlanRawPrice( state, plan.getProductId(), isMonthly )
+	);
+};
+
+export const planSlugToPlanProduct = createSelector( ( products, planSlug ) => {
+	const plan = getPlan( planSlug );
+	const planConstantObj = applyTestFiltersToPlansList( plan, abtest );
+	return {
+		planSlug,
+		plan: planConstantObj,
+		product: products[ planSlug ],
+	};
+} );
+
+export const computeFullAndMonthlyPricesForPlan = ( state, siteId, plan ) => ( {
+	priceFull: getDiscountedOrRegularPrice( state, siteId, plan, false ),
+	priceMonthly: getDiscountedOrRegularPrice( state, siteId, plan, true ),
```

Comment:
the way `getDiscountedOrRegularPrice()` is used here make me question why we call it twice. why not return the monthly and full price from every invocation of that call?

Reply:
This functions returns both monthly and a full price though :-) It's just a layer on top of `getDiscountedOrRegularPrice()` so we can reuse code instead of copying&pasting it